src/sentry/sentry_apps/api/serializers/app_platform_event.py

```diff
@@ -1,49 +1,93 @@
+from collections.abc import Mapping
 from time import time
+from typing import Any, Generic, TypedDict, TypeVar
 from uuid import uuid4
 
+from sentry.sentry_apps.models.sentry_app import SentryAppActionType, SentryAppResourceType
+from sentry.sentry_apps.models.sentry_app_installation import SentryAppInstallation
+from sentry.sentry_apps.services.app.model import RpcSentryAppInstallation
+from sentry.users.models.user import User
+from sentry.users.services.user import RpcUser
 from sentry.utils import json
 
 
-class AppPlatformEvent:
+class AppPlatformEventActor(TypedDict):
+    type: str
+    id: str | int
+    name: str
+
+
+class AppPlatformEventInstallation(TypedDict):
+    uuid: str
+
+
+T = TypeVar("T", bound=Mapping[str, Any])
+
+
+class AppPlatformEventBody(TypedDict, Generic[T]):
+    action: SentryAppActionType
+    installation: AppPlatformEventInstallation
+    data: T
+    actor: AppPlatformEventActor
+
+
+class AppPlatformEvent(Generic[T]):
     """
     This data structure encapsulates the payload sent to a SentryApp's webhook.
+
+    The data field is generic and should be typed with a TypedDict specified by the user.
     """
 
-    def __init__(self, resource, action, install, data, actor=None):
+    def __init__(
+        self,
+        resource: SentryAppResourceType,
+        action: SentryAppActionType,
+        install: RpcSentryAppInstallation | SentryAppInstallation,
+        data: T,
```

Comment:
Is there any association between the `data` type and the resource/action? Wondering if we can tie these together somehow so you don't have to remember to pass `T` in for data

Reply:
After consulting AI idk if there's a better way